src/ResourceManagement/MachineLearning/Microsoft.Azure.Management.MachineLearning/Management/WebServiceManagementClient.cs

```diff
@@ -0,0 +1,227 @@
+        /// </summary>
+        /// <param name="userCredentials">A users authenticated credentials.</param>
+        public WebServiceManagementClient(ServiceClientCredentials userCredentials)
+        {
+            _client = new AzureMLWebServicesManagementClient(userCredentials);
+            StudioWebServiceClient = new StudioServiceClient(userCredentials);
+        }
+
+        /// <summary>
+        /// WebServiceManagementClient constructor. Takes a subscription id.
+        /// </summary>
+        /// <param name="subscriptionId">The id for the user's subscription.</param>
+        public WebServiceManagementClient(string subscriptionId, ServiceClientCredentials userCredentials)
+        {
+            _client = new AzureMLWebServicesManagementClient(userCredentials);
+            StudioWebServiceClient = new StudioServiceClient(userCredentials);
+
+            _client.SubscriptionId = subscriptionId;
+        }
+
+        /// <summary>
+        /// Creates a web service object from a web service definition file. Also takes the name of the resource group.
+        /// </summary>
+        /// <param name="webServiceDefintionFilePath">The file path for the web service definition file. Expecting a JSON string within.</param>
+        /// <param name="resourceGroup">The resource group in which the web service is deployed.</param>
+        /// <returns></returns>
+        public WebService CreateWebServiceObject(string webServiceDefintionFilePath, string resourceGroup)
+        {
+            var ws = new WebService(webServiceDefintionFilePath, resourceGroup,_client);
+            return ws;
+        }
+
+        /// <summary>
+        /// Creates a web service object from an AutoRest generated WebService model. Also takes the name of the resource group.
+        /// </summary>
+        /// <param name="webService">The other web service from which we are creating a web service object. Serves as the object Definition internal state object.</param>
+        /// <param name="resourceGroup">The resource group in which the web service is deployed.</param>
+        /// <returns></returns>
+        public WebService CreateWebServiceObject(
+            Microsoft.Azure.Management.MachineLearning.WebServices.Models.WebService webService,
+            string resourceGroup)
+        {
+            var nWebService = new WebService(webService, resourceGroup,this._client);
+            return nWebService;
+        }
+
+        /// <summary>
+        /// Deploys the given web service in the resource group it specifies.
+        /// </summary>
+        /// <param name="webService">The web service to be deployed.</param>
+        /// <param name="force">Optional: If a web service with the same name exists, just go ahead and override it if true. Otherwise, throw an exception.</param>
+        public void DeployWebService(WebService webService, bool force = false)
+        {
+            if (!force)
```

Comment:
you could do some validation here, e.g. if ResourceGroup is empty or not? I'm asking this because I notice that when you generate a webservice from experiement, the resource group is empty. Other than fail the deployment later, you could catch it here. 

A better solution would be writing a method in WebService class called Validation(). Throw exception when something is wrong or missing. Then call it whereever you need it.